app/cmd/replica.go

```diff
@@ -73,6 +73,15 @@ func ReplicaCmd() cli.Command {
 				Value: "",
 				Usage: "Name of the replica instance (for validation purposes)",
 			},
+			cli.IntFlag{
+				Name:  "snapshot-max-count",
+				Value: 250,
+				Usage: "Maximum number of snapshots to keep",
+			},
+			cli.StringFlag{
+				Name:  "snapshot-max-size",
+				Usage: "Maximum total snapshot size in bytes or human readable 42kb, 42mb, 42gb",
```

Comment:
Does it mean that when:
1. User set the snapshot-max-size to 1GB
1. Write 2GB to the volume

Then cannot take the snapshot at all? Looks like a bit rough behavior. Could you provide the use-case for this feature? Thank you

Reply:
The use case is that we want to provide a way for user to control snapshot space. We offer `SnapshotMaxCount` and `SnapshotMaxSize` to manage it. For `SnapshotMaxCount`, the minimal value is `2`, because it means `2` disks in longhorn-engine. One is snapshot and another is volume head.

For `SnapshotMaxSize`:
* If `SnapshotMaxSize` is 0, it means there is no snapshot size limit for a Volume.
* If `SnapshotMaxSize` is not 0. The minimum value for `SnapshotMaxSize` is same as `Size * 2` in a Volume, because a volume can have at least two snapshots.


LEP: https://github.com/longhorn/longhorn/pull/6634

Reply:
Oh, so `SnapshotMaxSize` is the max value of sum(all snapshots' size) but not the max size of an individual snapshot. Got it!